### ISSUE TITLE
Storage Fee Improvement

### ISSUE THREAD

**Issue by bluesign** [onflow/flow#460]
## Instructions

Current implementation of storage fees puts responsibility of maintaining FLOW for storage on user. I guess this can have negative effects on adaptation. I think for end user, buying an NFT should cover the storage cost.

### Issue To Be Solved

I think if we think of NFTs (as size of a NFT can be determined depends on several factors such as holding content on chain or not) this will end up confusing from end user. 

### Suggest A Solution

I suggest some improvement especially on NFTs but also can be on any resource type, on creation on the resource, amount of FLOW needed to store the object that can be included in the resource. 

So basically if I am minting an NFT, I can include enough FLOW inside the NFT, and it can stay bound to that NFT until it is destroyed. (which in case of it is destroyed, balance can be deposited into owner that is destroying the NFT)

**Comment by ZayCodesLab** [comment]
+1

It's also worth noting that the current option of a platform providing Flow to an accounts  to cover storage costs can lead to security problems in that users can attempt to steal Flow from the platform that is attempting to provide a good UX to overcome the storage costs.

**Comment by aishairzay** [comment]
+1

This seems very important to prevent users from abusing platforms that are attempting to make good UX by sending flow to users that don't meet storage requirements.

> which in case of it is destroyed, balance can be deposited into owner that is destroying the NFT

I like this part of it as well. Provides some actual value to an NFT no matter what. But I think it should be configurable to allow for the Flow to end up elsewhere as well. I can imagine an nft where if you burn it, you're storage cost flow gets donated to a charity account or to the initial account that minted and covered this storage cost to begin with

**Comment by crash13override** [comment]
+1 to this! 
I think it would make things so much easier from a user perspective but also for developers when making gifts and airdrops to people that are new to the FLOW ecosystem.

**Comment by bluesign** [comment]
Thanks @crash13override, this issue just celebrated it’s first birthday yesterday. Probably will not happen unfortunately (due to some conflict to some Dapper strategy probably) 

Though it would be nice to have some feedback from Flow team 

Also as I am trying hard to get rid of meaningless storage fees, there is an alternative issue: https://github.com/onflow/flow/issues/791 which can solve the problem in an easier way.
